engine/runtime/src/main/java/org/enso/interpreter/runtime/util/CachingSupplier.java

```diff
@@ -55,6 +56,17 @@ public T get() {
     }
   }
 
+  /**
+   * Transform the result of this supplier by applying a mapping function {@code f}.
+   *
+   * @param f the mapping function
+   * @return the supplier providing the value with the mapping function applied
+   * @param <R> the result type
+   */
+  public <R> CachingSupplier<R> map(Function<T, R> f) {
```

Comment:
The _"tendency towards functional style"_ is clear...

Reply:
I, on the other hand, really like it :)

Reply:
There are [benchmark regressions](https://github.com/enso-org/enso/pull/11938#issuecomment-2565074927) and the only change that doesn't seem straightforward to me is this `map` method. But of course, I may be biased...